packages/protocol/contracts/governance/Governance.sol

```diff
@@ -649,7 +672,71 @@ contract Governance is IGovernance, Ownable, Initializable, UsingLockedGold, Ree
   }
 
   /**
-   * @notice Withdraws refunded Celo Gold commitments.
+   * @notice Whitelists the hash of a hotfix.
+   * @param txHash The abi encoded keccak256 hash of the hotfix transaction to be whitelisted.
+   * @param validatorIndex The index in the validator set of the whitelisting validator.
+   */
+  function whitelistHotfix(bytes32 txHash, uint256 validatorIndex) external nonReentrant {
+    if (msg.sender == approver) {
+      require(!hotfixes[txHash].approved);
+      hotfixes[txHash].approved = true;
+    } else {
+      IValidators validators = IValidators(registry.getAddressForOrDie(VALIDATORS_REGISTRY_ID));
+      require(msg.sender == validators.validatorAddressFromCurrentSet(validatorIndex));
+      require(!hotfixes[txHash].whitelisted[msg.sender]);
+      hotfixes[txHash].whitelisted[msg.sender] = true;
+    }
+
+    emit HotfixWhitelisted(txHash, msg.sender);
+  }
+
+  /**
+   * @notice Gives hotfix a prepared epoch for execution.
+   * @param txHash The hash of the hotfix to be prepared.
+   */
+  function prepareHotfix(bytes32 txHash) external {
```

Comment:
Why is this step necessary? Could we do this as part of execution?

Reply:
the `prepare` and `execute` phases are separated because of gas concerns
concretely, I was concerned about the unbounded iteration in checking the validator set whitelist in conflict with the unbounded iteration in executing the transaction

Reply:
additionally, if the merged `prepareAndExecute` reverts on the `isHotfixPassing` check because the validator quorum isn't reached, the underlying hotfix will be exposed